data/tabular/train_test_split.py

```diff
@@ -0,0 +1,228 @@
+            big_index_sets.append(index_set)
+        else:
+            small_index_sets.append(index_set)
+    random.seed(seed)
+    random.shuffle(big_index_sets)
+    random.shuffle(small_index_sets)
+    index_sets = big_index_sets + small_index_sets
+
+    if frac[2] == 0:
+        for index_set in index_sets:
+            if len(train) + len(index_set) <= train_size:
+                train += index_set
+                train_scaffold_count += 1
+            else:
+                val += index_set
+                val_scaffold_count += 1
+    else:
+        for index_set in index_sets:
+            if len(train) + len(index_set) <= train_size:
+                train += index_set
+                train_scaffold_count += 1
+            elif len(val) + len(index_set) <= val_size:
+                val += index_set
+                val_scaffold_count += 1
+            else:
+                test += index_set
+                test_scaffold_count += 1
+
+    return {
+        "train": df.iloc[train].reset_index(drop=True),
+        "valid": df.iloc[val].reset_index(drop=True),
+        "test": df.iloc[test].reset_index(drop=True),
+    }
+
+
+def rewrite_data_with_splits(
+    csv_paths: List[str],
+    train_test_df: pd.DataFrame,
+    override: bool = False,
+    check: bool = True,
+    repr_col: str = "SMILES",
+) -> None:
+    """Rewrite dataframes with the correct split column
+
+    Args:
+        csv_paths (List[str]): list of files to merge (data_clean.csv)
+        train_test_df (pd.DataFrame): dataframe containing merged SMILES representations
+            from all datasets uniquely split into train and test
+        override (bool): whether to override the existing data_clean.csv files
+            defaults to False
+        check (bool): whether to check if the split was successful
+            defaults to True. Can be turned off to save memory
+        repr_col (str): the column name for where SMILES representation is stored
+            defaults to "SMILES"
+    """
+    if check:
+        train_smiles = set(train_test_df.query("split == 'train'")["SMILES"].to_list())
+
+    for path in csv_paths:
+        read_dataset = pd.read_csv(path)
+        if repr_col in read_dataset.columns:
+            try:
+                read_dataset = read_dataset.drop("split", axis=1)
+                message = f"Split column found in {path}."
+                if override:
+                    message += " Overriding..."
+                print(message)
+            except KeyError:
+                print(f"No split column in {path}")
+
+            col_to_merge = "SMILES"
+            merged_data = pd.merge(
+                read_dataset, train_test_df, on=col_to_merge, how="left"
+            )
+            merged_data = merged_data.dropna()
+            if override:
+                merged_data.to_csv(path, index=False)
+            else:
+                merged_data.to_csv(path.replace(".csv", "_split.csv"), index=False)
```

Comment:
Copy `data_clean.csv` to something like `data_clean_default_split.csv` and save the new split info in `data_clean.csv`. Like that we don't have to rename the files or adapt the logic in the other scripts that are based on it.